test/integration/narrow/routes/vet/reference.test.js

```diff
@@ -85,8 +88,8 @@ describe('Vet, enter reference test', () => {
       expect($('#reference').val()).toEqual(expectedVal)
     })
 
-    // TODO: remove skip when application lookup functionality plugged in
-    test.skip('returns 404 when payload is valid', async () => {
+    messaging.receiveMessage = jest.fn().mockReturnValueOnce(null).mockReturnValue({ applicationId: 'VV-1234-5678' })
```

Comment:
Can this be moved closer to the initialisation of the messaging mock please? And possibly use the `reference` variable for the `applicationId`.
It'll be easier to see the usage of it and the mocks get hoisted anyway so it makes it closer to the execution model.